### PR TITLE
8320715: Improve the tests of test/hotspot/jtreg/compiler/intrinsics/float16

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you review the patch to improve the tests of test/hotspot/jtreg/compiler/intrinsics/float16?
Thanks.

Currently, there are several areas where improvements can be made:
1. test should not exit early before all test cases have run.
2. some output hide the data details, which should be printed out to help debug any potential issues, e.g. currently it could output some log like below in case of wrong implementation of a ConvF2HF intrinsic:
`Inconsistent result for Float.floatToFloat16(NaN/ff802000): fc01 != fc01`, which is rather confusing, not helpful. Suggested output would be: `Inconsistent result for Float.floatToFloat16(NaN/ff802000): fc01(64513) != fc01(-1023)`

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8320715](https://bugs.openjdk.org/browse/JDK-8320715): Improve the tests of test/hotspot/jtreg/compiler/intrinsics/float16 (**Enhancement** - P4)

### Reviewers
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16815/head:pull/16815` \
`$ git checkout pull/16815`

Update a local copy of the PR: \
`$ git checkout pull/16815` \
`$ git pull https://git.openjdk.org/jdk.git pull/16815/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16815`

View PR using the GUI difftool: \
`$ git pr show -t 16815`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16815.diff">https://git.openjdk.org/jdk/pull/16815.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16815#issuecomment-1826750798)

### CODE REVIEW

File: test/hotspot/jtreg/compiler/intrinsics/float16/Binary16Conversion.java [hunk 1]
```diff
@@ -326,11 +327,12 @@ private static int testAlternativeImplementation() {
...
             float f = Float.intBitsToFloat((int)ell);
             short s1 = Float.floatToFloat16(f);
-            short s2 =    testAltFloatToFloat16(f);
+            short s2 = testAltFloatToFloat16(f);
 
             if (s1 != s2) {
                 errors++;
-                System.out.println("Different conversion of float value " + Float.toHexString(f));
+                System.out.println("Different conversion of float value 1 (" + f + "/" + Integer.toHexString(Float.floatToRawIntBits(f)) + "): " +
+                                    Integer.toHexString(s1 & 0xffff) + "(" + s1 + ")" + " != " + Integer.toHexString(s2 & 0xffff) + "(" + s2 + ")");
```
Comment (vnkozlov) [inline test/hotspot/jtreg/compiler/intrinsics/float16/Binary16Conversion.java:335]: May be split this long line as you did in previous case.
Why you want `1` in `value 1`?

**Review by vnkozlov — COMMENTED** [review]
Good improvement. I have comment.

File: test/hotspot/jtreg/compiler/intrinsics/float16/Binary16Conversion.java (hunk 1 above)
Comment (Hamlin-Li) [reply test/hotspot/jtreg/compiler/intrinsics/float16/Binary16Conversion.java]: > May be split this long line as you did in previous case.
Done

> Why you want 1 in value 1?
Oh, good catch, it's a typo, fixed.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by Hamlin-Li** [comment]
@vnkozlov Thanks for your reviewing.

/integrate

**Pull request closed without merging**